### PR TITLE
Add 6 new message types

### PR DESCRIPTION
**Pull request by NCPlayz**
### Summary

Adds:
```
    guild_stream                                 = 13
    guild_discovery_disqualified                 = 14
    guild_discovery_requalified                  = 15
    guild_discovery_grace_period_initial_warning = 16
    guild_discovery_grace_period_final_warning   = 17
    thread_created                               = 18
```

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -911,6 +911,40 @@ of :class:`enum.Enum`.
         The system message denoting that an announcement channel has been followed.
 
         .. versionadded:: 1.3
+    .. attribute:: guild_stream
+
+        The system message denoting that a member is streaming in the guild.
+
+        .. versionadded:: 1.4
```
Comment (PikalaxALT) [inline docs/api.rst:918]: Bump to 1.6

File: docs/api.rst [hunk 2]
```diff
@@ -911,6 +911,40 @@ of :class:`enum.Enum`.
...
+
+        The system message denoting that the guild has failed to meet the Server
+        Discovery requirements for 3 weeks in a row.
+    
+        .. versionadded:: 1.4
+    .. attribute:: thread_created
+
+        The system message denoting that a member has started a thread.
+
+        .. versionadded:: 1.4
```
Comment (PikalaxALT) [inline docs/api.rst:947]: Document the reply type here

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Comment by NCPlayz** [comment]
Thanks for the review @PikalaxALT, this PR was mainly put on hold due to the inclusion of the thread message type (along with #5201) which hasn't become official yet, so I might move thread related features to another PR and bump this one since the reply type is a priority at the moment.

**Comment by NCPlayz** [comment]
Removed thread and reply message types as they require an API version bump to v8. The rest of the types still present can be gracefully merged into 1.7.

**Pull request merged**